lib/Fmt_ast.ml

```diff
@@ -996,11 +996,11 @@ and fmt_pattern c ?pro ?parens ({ctx= ctx0; ast= pat} as xpat) =
           @@
           match ppat_desc with
           | Ppat_var {txt= txt'; _}
-            when field_alias ~field:lid1.txt (Longident.parse txt')
+            when field_alias ~field:lid1.txt (Longident.parse txt' [@ocaml.warning "-3"])
```

Comment:
Can you replace the `Longindent.parse` calls to a new function `Migrate_ast.Parse.longident`? that way there's only one place where we suppress the warning. You can try to replace it with `Parse.longindent` (the one from compiler-libs) or `unflatten` after splitting with `String.split ~on:"."` but that can be done in a separate PR, after or before this one.